exporter/containerimage/export.go

```diff
@@ -0,0 +1,228 @@
+				lower, err = parent.Mount(ctx)
+				if err != nil {
+					return nil, err
+				}
+			}
+			upper, err := ref.Mount(ctx)
+			if err != nil {
+				return nil, err
+			}
+			descr, err := e.opt.Differ.DiffMounts(ctx, lower, upper, ocispec.MediaTypeImageLayer, ref.ID())
+			if err != nil {
+				return nil, err
+			}
+			if err := e.blobmap.SetBlob(ctx, ref.ID(), descr.Digest); err != nil {
+				return nil, err
+			}
+			return diffPair{diffID: descr.Digest, blobsum: descr.Digest}, nil
+		})
+		if err != nil {
+			return err
+		}
+		currentPair = dp.(diffPair)
+		return nil
+	})
+	err := eg.Wait()
+	if err != nil {
+		return nil, err
+	}
+	return append(append([]diffPair{}, diffPairs...), currentPair), nil
+}
+
+type imageExporterInstance struct {
+	*imageExporter
+	targetName string
+}
+
+func (e *imageExporterInstance) Export(ctx context.Context, ref cache.ImmutableRef) error {
+	diffPairs, err := e.getBlobs(ctx, ref)
+	if err != nil {
+		return err
+	}
+
+	diffIDs := make([]digest.Digest, 0, len(diffPairs))
+	for _, dp := range diffPairs {
+		diffIDs = append(diffIDs, dp.diffID)
+	}
+
+	dt, err := json.Marshal(imageConfig(diffIDs))
+	if err != nil {
+		return errors.Wrap(err, "failed to marshal image config")
+	}
+
+	dgst := digest.FromBytes(dt)
+
+	if err := content.WriteBlob(ctx, e.opt.ContentStore, dgst.String(), bytes.NewReader(dt), int64(len(dt)), dgst); err != nil {
+		return errors.Wrap(err, "error writing config blob")
+	}
+
+	mfst := ocispec.Manifest{
+		Config: ocispec.Descriptor{
+			Digest:    dgst,
+			Size:      int64(len(dt)),
+			MediaType: ocispec.MediaTypeImageConfig,
+		},
+	}
+	mfst.SchemaVersion = 2
+
+	for _, dp := range diffPairs {
+		info, err := e.opt.ContentStore.Info(ctx, dp.blobsum)
+		if err != nil {
+			return errors.Wrapf(err, "could not get blob %s", dp.blobsum)
+		}
+		mfst.Layers = append(mfst.Layers, ocispec.Descriptor{
+			Digest:    dp.blobsum,
+			Size:      info.Size,
+			MediaType: ocispec.MediaTypeImageLayerGzip,
+		})
+	}
+
+	dt, err = json.Marshal(mfst)
+	if err != nil {
+		return errors.Wrap(err, "failed to marshal manifest")
+	}
+
+	dgst = digest.FromBytes(dt)
+
+	if err := content.WriteBlob(ctx, e.opt.ContentStore, dgst.String(), bytes.NewReader(dt), int64(len(dt)), dgst); err != nil {
+		return errors.Wrap(err, "error writing manifest blob")
+	}
+
+	logrus.Debugf("wrote manifest %s", dgst)
+
+	if e.opt.Images != nil && e.targetName != "" {
+		e.opt.Images.Update(ctx, e.targetName, ocispec.Descriptor{
+			Digest:    dgst,
+			Size:      int64(len(dt)),
+			MediaType: ocispec.MediaTypeImageManifest,
+		})
+		logrus.Debugf("updated tag for %s", e.targetName)
+	}
+
+	return err
+}
+
+// this is temporary: should move to dockerfile frontend
+func imageConfig(diffIDs []digest.Digest) ocispec.Image {
+	img := ocispec.Image{
+		Architecture: runtime.GOARCH,
+		OS:           runtime.GOOS,
+	}
+	img.RootFS.Type = "layers"
+	img.RootFS.DiffIDs = diffIDs
+	img.Config.WorkingDir = "/"
```

Comment:
I _think_ this would work in windows as well. Also, `Dockerfile` support that should remove this function will likely come before windows support. With the current exporter there isn't a good way to determine the platform for image and using the current runtime platform may not be correct.